### ISSUE TITLE
How to compile it in melodic?

### ISSUE THREAD

**Issue by Zcyyy** [VIS4ROB-lab/ccm_slam#19]
Hi,
My environment is ubuntu 18.04 with ros-melodic.When i try to compile use catkin build,it always crash at 58%.I don't know what happened.I have tried several times.

**Comment by manthan99** [comment]
Hey @Zcyyy 
I just tried out compiling with 18.04 and ros-melodic. Mine is able to compile. Could you please post the exact error which you are getting.
Just did a small change in the CMakeList of cslam package in line no. 16. Just replace kinetic with melodic inline 16.
`elseif($ENV{ROS_DISTRO} MATCHES "melodic")`

**Comment by patriksc** [comment]
@manthan99 Thanks for sharing your experience.
@Zcyyy I've also just tested it on my machine, following exactly the manual - I don't have any issues compiling, apart from the check against ROS kinetic or indigo. I've adjusted the CMakeLists accordingly, should be fine now. 
Did you make sure you were following all of the steps? E.g. that the workspace is properly initialized and extended to ROS melodic, DBoW2 and g2o are built, etc.?
